addons/website_sale/controllers/main.py

```diff
@@ -265,41 +247,37 @@ def shop(self, page=0, category=None, search='', ppg=False, **post):
 
     @http.route(['/shop/product/<model("product.template"):product>'], type='http', auth="public", website=True)
     def product(self, product, category='', search='', **kwargs):
-        cr, uid, context, pool = request.cr, request.uid, request.context, request.registry
-        category_obj = pool['product.public.category']
-        template_obj = pool['product.template']
+        local_context = dict(request.env.context or {})
```

Comment:
are you sure the context can be None? Isn't `local_context = dict(request.env.context, active_id=product.id)` enough?